NServiceBus.RavenDB Upgrade Guide

colin-higgins opened the pull request:
@Particular/tf-async-persistence 

Putting together the skeleton and snippets for upgrade information which is relevant to users.
Thoughts?

Some of this stuff might be duplicated in the main upgrade document. I tried to pull that stuff out, but more eyes would be good.

SimonCropp commented on nservicebus/upgrades/5to6/raven-upgrade.md:
```diff
@@ -0,0 +1,47 @@
+---
+title: Upgrade Raven from NServiceBus 5 to 6
```
shouldnt this reflect the ravendb numbers not the nsb numbers?

colin-higgins replied:
I was wondering about that and intentionally chose NServiceBus numbers. I feel like anyone looking into this upgrade guide would only be doing this because they are jumping from 5 to 6. They in fact could not if they weren't. 

You don't think we should be uniform in the upgrade guides by using core's major?

SimonCropp commented on nservicebus/upgrades/5to6/raven-upgrade.md:
```diff
@@ -0,0 +1,47 @@
+
+## ISessionProvider is obsolete
+
+In previous versions of NServiceBus.RavenDB, to gain access to a shared raven session you had to request `ISessionProvider` as a constructor argument. We have deprecated this interface in favor of `SynchronizedStorageSession` on `IMessageHandlerContext`.
```
remove `you` and `we`

SimonCropp commented on nservicebus/upgrades/5to6/raven-upgrade.md:
```diff
@@ -0,0 +1,47 @@
+
+## SynchronizedStorageSession
+
+To gain access to your shared session, call the `Session()` extension method on `Synchronized storage session.
```
remove `your`

SimonCropp commented on nservicebus/upgrades/5to6/raven-upgrade.md:
```diff
@@ -0,0 +1,47 @@
+
+### Session is available regardless of features enabled
+
+In version 5, the `RavenStorageSession` was only registered if at least one out of `Outbox` and `Sagas` were enabled. There are possible use cases for using the NServiceBus wrapped session to talk to RavenDB so we removed the prerequisites.
```
this should be the nsb.raven number not the nsb number

SimonCropp commented on Snippets/Snippets_6/UpgradeGuides/5to6/SharedSessionEndpointConfig.cs:
```diff
@@ -0,0 +1,19 @@
+    {
+        public async Task DoStuff(BusConfiguration configuration, IAsyncDocumentSession someAsyncSession)
+        {
+            #region 5to6-ravensharedsession
```
this is not 5to6

colin-higgins commented:
@SimonCropp @Particular/tf-async-persistence 

Updated the snippets and fixed the versions. How should we include this in the upgrade guide, just another link?

colin-higgins commented on nservicebus/upgrades/5to6/raven-upgrade.md:
```diff
@@ -0,0 +1,41 @@
+
+### For multiple correlations, use a custom finder
+
+Lorem ipsum...
```
Still TODO obviously

SimonCropp commented on Snippets/Snippets_5/UpgradeGuides/Downstream/Raven/3to4/HandlerWithRavenSession.cs:
```diff
@@ -0,0 +1,25 @@
+    #region 3to4-acccessingravenfromhandler
+    public class HandlerWithRavenSession : IHandleMessages<MyMessage>
+    {
+        private ISessionProvider sessionProvider;
```
remove the private

SimonCropp commented on Snippets/Snippets_5/UpgradeGuides/Downstream/Raven/3to4/HandlerWithRavenSession.cs:
```diff
@@ -0,0 +1,25 @@
+﻿namespace Snippets6.UpgradeGuides._5to6
```
me thinks the namespace is wrong?

colin-higgins replied:
Haha

SimonCropp commented on nservicebus/upgrades/5to6/raven-upgrade.md:
```diff
@@ -0,0 +1,41 @@
+
+## Namespace change
+
+The `RavenDBPersistence` has moved from `NServiceBus.Persistence` namespace to `NServiceBus`. There is no need to have an extra using statement if you have the NuGet package installed. 
```
remove you

SimonCropp commented on nservicebus/upgrades/5to6/raven-upgrade.md:
```diff
@@ -0,0 +1,41 @@
+
+## Configuring a shared session
+
+Configuring a shared raven session now requires a `Func<IAsyncDocumentSession>` instead of a `Func<IDocumentSession`. 
```
missing `>`

SimonCropp commented on nservicebus/upgrades/5to6/raven-upgrade.md:
```diff
@@ -0,0 +1,41 @@
+
+## Saga correlation properties
+
+In NServiceBus.RavenDB, more than one correlation property was never allowed. In NServiceBus 6, this has become the default for all persistence implementations.
```
if no change is required here do we even need to mention it?

colin-higgins replied:
I was thinking about that as well. It seemed like a nice thing to know, but maybe it doesn't matter. I'll remove it.

SimonCropp commented on Snippets/Snippets_5/UpgradeGuides/Downstream/Raven/3to4/SharedSessionEndpointConfig.cs:
```diff
@@ -0,0 +1,19 @@
+﻿namespace Snippets6.UpgradeGuides._5to6
```
not 5to6
